rivershared/util/randutil/rand_util.go

```diff
@@ -2,31 +2,14 @@ package randutil
 
 import (
 	cryptorand "crypto/rand"
-	"encoding/binary"
 	"encoding/hex"
-	mathrand "math/rand"
-	"sync"
+	"math/rand/v2"
 	"time"
 )
 
-// NewCryptoSeededConcurrentSafeSource generates a new pseudo-random source
-// that's been created with a cryptographically secure seed to ensure reasonable
-// distribution of randomness between nodes and services, and wrapped so that
-// access to it is concurrent safe.
-//
-// This project uses this technique instead of falling back on crypto/rand
-// because uses of randomness don't need to be cryptographically secure, and the
-// non-crypto variant is about twenty times faster.
-func NewCryptoSeededConcurrentSafeRand() *mathrand.Rand {
-	return mathrand.New(newCryptoSeededConcurrentSafeSource())
-}
-
 // DurationBetween generates a random duration in the range of [lowerLimit, upperLimit).
-//
-// TODO: When we drop Go 1.21 support, switch to `math/rand/v2` and kill the
-// `rand.Rand` argument.
-func DurationBetween(rand *mathrand.Rand, lowerLimit, upperLimit time.Duration) time.Duration {
-	return time.Duration(IntBetween(rand, int(lowerLimit), int(upperLimit)))
+func DurationBetween(lowerLimit, upperLimit time.Duration) time.Duration {
```

Comment:
riverpro will need a bump to correspond with these internal API changes

Reply:
Makes sense. I'll send one over.